version: "3.8"

networks:
  metrics_collection_network:
    external: true

volumes:
  grafana_data:

secrets:
  grafana_admin_user:
    external: true
  grafana_admin_password:
    external: true

services:
  grafana:
    # image: portainer/template-swarm-monitoring:grafana-9.5.2
    image: grafana/grafana:latest
    ports:
      - ${GRAFANA_PORT?}:3000
    volumes:
      - grafana_data:/var/lib/grafana
    secrets:
      - grafana_admin_user
      - grafana_admin_password
    environment:
      - GF_SECURITY_ADMIN_USER__FILE=/run/secrets/grafana_admin_user
      - GF_SECURITY_ADMIN_PASSWORD__FILE=/run/secrets/grafana_admin_password
      - GF_USERS_ALLOW_SIGN_UP=false
    networks:
      - metrics_collection_network
    deploy:
      replicas: 1
      restart_policy:
        condition: on-failure
        delay: 5s
      resources:
        limits:
          cpus: "0.5"
          memory: 512M
        reservations:
          cpus: "0.2"
          memory: 256M
      update_config:
        parallelism: 1
        order: stop-first
        failure_action: rollback
      